### PR TITLE
Added exception catching to send default email if template file raises any exception

### LINKED ISSUE #24919: Send default email if file "html_content_template" not found
### Apache Airflow version

2.3.2 (latest released)

### What happened

I created a new email template to be sent when there are task failures. I accidentally added the path to the `[email] html_content_template` and `[email] subject_template` with a typo and no email was sent. The task's log is the following:

```
Traceback (most recent call last):
  File "/home/user/.conda/envs/airflow/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1942, in handle_failure
    self.email_alert(error, task)
  File "/home/user/.conda/envs/airflow/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 2323, in email_alert
    subject, html_content, html_content_err = self.get_email_subject_content(exception, task=task)
  File "/home/user/.conda/envs/airflow/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 2315, in get_email_subject_content
    subject = render('subject_template', default_subject)
  File "/home/user/.conda/envs/airflow/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 2311, in render
    with open(path) as f:
FileNotFoundError: [Errno 2] No such file or directory: '/home/user/airflow/config/templates/email_failure_subject.tmpl'
```

I've looked the TaskInstance class (https://github.com/apache/airflow/blob/main/airflow/models/taskinstance.py).

I've seen that the `render` function (https://github.com/apache/airflow/blob/bcf2c418d261c6244e60e4c2d5de42b23b714bd1/airflow/models/taskinstance.py#L2271) has a `content` parameter, which is not used inside.

I guess the solution to this bug is simple: just add a `try - catch` block and return the default content in the `catch` part.

### What you think should happen instead

_No response_

### How to reproduce

_No response_

### Operating System

CentOS Linux 8

### Versions of Apache Airflow Providers

_No response_

### Deployment

Other

### Deployment details

Conda environment

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by ecodina**
This PR closes #24919 

A try - except block has been added to prevent email failure when a custom template file is used for task error notifications.

Although once Airflow has been correctly configured we should never need the except block, sometimes we may encounter issues when updating the template file. I believe Airflow should always send the email and, in case the template file has any problems, it should just send the default template.

### CODE REVIEW

**Comment by ecodina** [comment]
A test failed, but I'd like to argue my case:

In this try - except we will mostly catch the FileNotFoundError, which is explicit.

In any other case (e.g. an IO error), I believe we should still send the email with the default template to make sure we are always notified.

File: airflow/models/taskinstance.py [hunk 1]
```diff
@@ -2271,8 +2271,13 @@ def get_email_subject_content(
...
                 if conf.has_option('email', key):
                     path = conf.get_mandatory_value('email', key)
-                    with open(path) as f:
-                        content = f.read()
+                    try:
+                        with open(path) as f:
+                            content = f.read()
+                    except FileNotFoundError:
+                        self.log.warning(f"Could not find email template file '{path}'. Using defaults...")
+                    except:
```
Comment (uranusjr) [inline airflow/models/taskinstance.py]: -1 to this blank catch. It’s probably worthwhile to catch all OSError, but this is way to broad.

File: airflow/models/taskinstance.py [hunk 2]
```diff
@@ -2271,8 +2271,13 @@ def get_email_subject_content(
             def render(key: str, content: str) -> str:
                 if conf.has_option('email', key):
                     path = conf.get_mandatory_value('email', key)
-                    with open(path) as f:
-                        content = f.read()
+                    try:
+                        with open(path) as f:
+                            content = f.read()
+                    except FileNotFoundError:
+                        self.log.warning(f"Could not find email template file '{path}'. Using defaults...")
```
Comment (uranusjr) [inline airflow/models/taskinstance.py]: Better to use `{path!r}`

**Comment by ecodina** [comment]
Follwing @uranusjr advice, I replaces the broad exception to only catch OSError (which is the only one that should happen, actually). I have also added a UnitTest:

`tests/models/test_taskinstance.py::TestTaskInstance::test_email_alert_with_filenotfound_config PASSED `

With this changes, the PR should be ready to be merged.

**Review by uranusjr — APPROVED** [review]

**Comment by potiuk** [comment]
Only static checks already fixed in main are failing. MErging.

**Review by potiuk — APPROVED** [review]

**Pull request merged**